CMakeLists.txt

```diff
@@ -122,8 +122,13 @@ if(NOT CUKE_DISABLE_QT)
     find_package(Qt5Test)
 
     if(${Qt5Core_FOUND} AND ${Qt5Widgets_FOUND} AND ${Qt5Test_FOUND})
+        set(QT5_FOUND true)
         message(STATUS "Found Qt version: ${Qt5Core_VERSION_STRING}")
         set(QT_LIBRARIES Qt5::Core Qt5::Widgets Qt5::Test)
+        if(NOT ${Qt5Core_VERSION_STRING} VERSION_LESS 5.7)
+            message(STATUS "C++11 is needed from Qt version 5.7.0, building with c++11 enabled")
+            set(CMAKE_CXX_STANDARD 11)
```

Comment:
This precludes building as C++14 or 17. You could instead use `target_compile_features(cucumber-cpp PUBLIC ${some_CXX11_features_that_Qt_5_7_uses})` where that list of features that you require are a subset of [CMAKE_CXX_KNOWN_FEATURES](https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html).

Or an easier approach may be to write it like this to prevent overriding the user-requested C++ version:
```cmake
if(NOT Qt5Core_VERSION_STRING VERSION_LESS 5.7
    AND (NOT DEFINED CMAKE_CXX_STANDARD OR NOT CMAKE_CXX_STANDARD STREQUAL 98))
```